src/state/state_api.rs

```diff
@@ -120,5 +122,5 @@ pub trait State {
     fn get_sierra_program(
         &mut self,
         class_hash: &ClassHash,
-    ) -> Result<Vec<BigUintAsHex>, StateError>;
+    ) -> Option<Arc<(SierraProgram, ContractEntryPoints)>>;
```

Comment:
why an Option instead of a Result? I think we should take into account that multiple implementations can find handy to return a Result to represent other kind of errors.

Reply:
Because of the function's behavior: either we have it or we don't. If the error is always `we don't have it` then we don't need an error. But this API has been removed from there because of the contract cache.